### ISSUE TITLE
Error should be reported if it is possible to instantiate a class without calling super

### ISSUE THREAD

**Issue by ddadon10** [microsoft/TypeScript#53460]
# Bug Report
By putting a `if/else` block in a constructor it is possible to compile a class that will not call `super()` and therefore fail at runtime with this error:
` Must call super constructor in derived class before accessing 'this' or returning from derived constructor `

In other situation, Typescript is preventing this kind of mistake by raising an error at compile time like:
`Constructors for derived classes must contain a 'super' call.(2377)`

### 🔎 Search Terms
constructor, super, error

### 🕗 Version & Regression Information

5.0.2, Nightly.
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about  [Common "Bugs" That Aren't Bugs](https://github.com/Microsoft/TypeScript/wiki/FAQ#common-bugs-that-arent-bugs) 

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?useDefineForClassFields=true&ts=5.0.2#code/MYGwhgzhAEBiD29oG8BQ1oDNEC5oQBcAnASwDsBzdaYeMwogV2APiIApt48HyKBKFNQwEAFiQgA6LtAC8WRNQC+qFalCQYAITBFoAUwAeBfWQAmMBEjQZa9Ys1YcARrrxlGAW2f6i-GxjQJJjQ7K56spHQAEyCAYH4jAAOvuwARNFp-MLQKhgqanaE0OFy0GT6AO7QOhwAjNnh0oiSwKK6AIIE7AAM-EA)

### 💻 Code
```ts
class Foo {
  foo: string
  constructor(foo: string) {
    this.foo = foo
  }
}

class Bar extends Foo {
  constructor(bar: number){
    if (bar === 2) {
      super("2")
    }
  // Super is not called if bar is not 2
  }
}

const bar = new Bar(1) // Fail at runtime with: Must call super constructor in derived class before accessing 'this' or returning from derived constructor 
bar.foo.charAt(0)
```

### 🙁 Actual behavior
 Fail at runtime with: `Must call super constructor in derived class before accessing 'this' or returning from derived constructor `
### 🙂 Expected behavior
Should fail at compile time with: `Constructors for derived classes must contain a 'super' call.(2377)`

**Comment by fatcerberus** [comment]
I feel like this is a duplicate…
